### ISSUE TITLE
maxx - 

### ISSUE THREAD

**Issue by 5xcor** [GregoryMorse/trendln#3]
Hi,

this lib is a very good piece. Thanks for the sharing.

I found the bug:
but in the row 699 is var maxx - is not declared.

Milan

**Comment by GregoryMorse** [comment]
Can you tell me the Python version you are using?  It is strange because the `for` loop scope in Python is supposed to leak outside the `for` block.  All discussion I find shows that the code should work as it does when I test it.

I suppose I will rewrite it as a `while` loop to be on the safe side anyway as its a poor coding style regardless.  But I would like to know the details of why regardless.

**Comment by 5xcor** [comment]
I use python 3.7.4

    def add_trend(trend, lbl, clr, bFirst):
        for ln in trend[:numbest]:
            maxx=1. ## - add?
            for maxx in range(ln[0][-1]+1, len_h):
                ypred = ln[1][0] * maxx + ln[1][1]
                if (hist[maxx] > ypred and hist[maxx-1] < ypred or hist[maxx] < ypred and hist[maxx-1] > ypred or
                    ypred > max_h + (max_h-min_h)*pctbound or ypred < min_h - (max_h-min_h)*pctbound): break
            x_vals = np.array((ln[0][0], maxx)) # plt.gca().get_xlim()) # maxx is missing
            y_vals = ln[1][0] * x_vals + ln[1][1]
            if bFirst:
                plt.plot([ln[0][0], maxx], y_vals, clr, label=lbl)
                bFirst = False
            else: plt.plot([ln[0][0], maxx], y_vals, clr)
        return bFirst

**Comment by GregoryMorse** [comment]
Nice bug find.  All Python versions behave the same - but - apparently the range is empty so ln[0][-1]+1 >= len_h.  So maxx is never set.  Its legitimately a bug.  Better to use a while loop and I will include this in next release, thanks for this report!

```
            maxx = ln[0][-1]+1
            while maxx < len_h:
                ypred = ln[1][0] * maxx + ln[1][1]
                if (hist[maxx] > ypred and hist[maxx-1] < ypred or hist[maxx] < ypred and hist[maxx-1] > ypred or
                    ypred > max_h + (max_h-min_h)*pctbound or ypred < min_h - (max_h-min_h)*pctbound): break
                maxx += 1
            x_vals = np.array((ln[0][0], maxx)) # plt.gca().get_xlim())
```

**Comment by GregoryMorse** [comment]
Your could could also work but not with `maxx=1. ## - add?` as it should reflect the initial value of `maxx = ln[0][-1]+1`.  But the while loop is cleaner in this case since it avoids reassignment of the variable on the first iteration of the `for` loop.

**Comment by GregoryMorse** [comment]
Fixed in 0.1.10
